openstates/mi/people.py

```diff
@@ -136,25 +138,27 @@ def scrape_upper(self, chamber):
             # are on the page linked off "Contact Me"
 
             # data has a typo in a row
+            email = None
             contact_url = [
                 a for a in row.xpath(".//a")
                 if a.text in ('Contact Me', 'Conact Me')][0].get('href')
-            contact_html = self.get(contact_url).text
-            contact_doc = lxml.html.fromstring(contact_html)
-
-            email = None
-            header_email = contact_doc.xpath("//a[@class='header_email']")
-            if header_email:
-                email = header_email[0].text
-            else:
-                # not using the most common template, but maybe they
-                # dropped their email on the page somewhere
-                links = contact_doc.xpath('//a') or []
-                text_email = [a for a in links
-                              if 'mailto:' in (a.get('href') or '')]
-                if text_email:
-                    email = text_email[0].text
-
+            try:
+                contact_html = self.get(contact_url).text
+                contact_doc = lxml.html.fromstring(contact_html)
+
+                header_email = contact_doc.xpath("//a[@class='header_email']")
+                if header_email:
+                    email = header_email[0].text
+                else:
+                    # not using the most common template, but maybe they
+                    # dropped their email on the page somewhere
+                    links = contact_doc.xpath('//a') or []
+                    text_email = [a for a in links
+                                  if 'mailto:' in (a.get('href') or '')]
+                    if text_email:
+                        email = text_email[0].text
+            except requests.exceptions.TooManyRedirects:
+                print("Conatct Link Not Working for %s" % name)
```

Comment:
Instead of `print("message")`, can you use `self.warning("message")`? This uses more formalized Python `logging`.

Reply:
Also, small typo in the error message itself.